src/app/components/ScheduleShiftForm.tsx

```diff
@@ -10,59 +9,28 @@ import {
 import * as Validators from '../infrastructure/Validators';
 import * as DateTimeFieldConst from './FormElements/DateTimeFieldConst';
 import {
-    DateType,
-    DaysOfWeek
+    DateType
 } from '../api/Api';
 import DaysOfWeekChecklist from './FormElements/DaysOfWeekChecklist';
 import WorkSectionSelector from './FormElements/WorkSectionSelector';
 import NumberSpinner from './FormElements/NumberSpinner';
-import CheckboxField from './FormElements/CheckboxField';
 
 export interface ScheduleShiftFormProps {
     handleSubmit?: () => void;
     onSubmitSuccess?: () => void;
-    weekStart: DateType;
+    weekStart?: DateType;
+    isSingleShift?: boolean;
+    shiftTitle?: string;
 }
 
 export default class ScheduleShiftForm extends
     React.Component<ScheduleShiftFormProps & InjectedFormProps<{}, ScheduleShiftFormProps>, {}> {
 
-    public static createDefaultShift() {
-        return {
-            startTime: moment().hour(9).minute(0),
-            endTime: moment().hour(17).minute(0),
-            days: DaysOfWeek.Weekdays,
-            repeatNumber: 1
-        };
-    }
-
-    render() {
-        const { handleSubmit } = this.props;
-        return (
-            <div>
-                <Form onSubmit={handleSubmit}>
-                    <Field
-                        name="workSectionId"
-                        component={WorkSectionSelector}
-                        label="Work Section"
-                    />
-                    <Field
-                        name="workSectionNotRequired"
-                        component={CheckboxField}
-                        label="Work Section Not Applicable"
-                    />
-                    <Field
-                        name="startTime"
-                        component={DateTimeFieldConst.TimeField}
-                        label="Start Time"
-                        validate={[Validators.required]}
-                    />
-                    <Field
-                        name="endTime"
-                        component={DateTimeFieldConst.TimeField}
-                        label="End Time"
-                        validate={[Validators.required]}
-                    />
+    private renderMultiShiftCreationFields() {
```

Comment:
I wonder if you just combine these two into the render method, since they are mutually exclusive:

i.e.

```ts
render(){
return (
  {isSingleShift && (
   // single shift fields
  )}
  {!isSingleShift && (
    // multishift fields
  )}
)
}
```

Rather than always calling them but returning `''` if not applicable

Reply:
They aren't totally mutually exclusive - if it is a shift creation, we still need the shiftFields, but we also need the multishift fields